chain/chain/src/chain.rs

```diff
@@ -595,13 +600,17 @@ impl Chain {
         &mut self,
         tries: ShardTries,
         height: BlockHeight,
+        mut gc_blocks_remaining: NumBlocks,
```

Comment:
Is it intended that we take ownership here instead of using a reference? It seems that this implies we will garbage collect more than intended.

Reply:
Awesome.